tools/evmc/skip_space_iterator.hpp

```diff
@@ -0,0 +1,60 @@
+// EVMC: Ethereum Client-VM Connector API.
+// Copyright 2022 The EVMC Authors.
+// Licensed under the Apache License, Version 2.0.
+#pragma once
+
+#include <iterator>
+
+namespace evmc
+{
+/// The constexpr variant of std::isspace().
+inline constexpr bool isspace(char ch) noexcept
+{
+    // Implementation taken from LLVM's libc.
+    return ch == ' ' || (static_cast<unsigned>(ch) - '\t') < 5;
+}
+
+/// The input filter iterator which skips whitespace characters from the base input iterator.
+template <typename BaseIterator>
+struct skip_space_iterator
```

Comment:
We extract the filter function (`isspace`) to get generic filter iterator (like `boost::filter_iterator`). This can be done later to filter out literal separator (`'`).